R/liger.R

```diff
@@ -1941,88 +1941,102 @@ SNF.liger <- function(
+      "Query datasets:\n",
+      paste("  ", as.character(queries), "\n")
+      )
   }
   else { # only given query datasets
-    queries = as.list(queries)
-    if (reference %in% queries){
-      stop('Invalid query datasets: Reference dataset CANNOT be inclued in the query datasets')
+    queries <- as.list(queries)
+    if (reference %in% queries) {
+      stop("Invalid query datasets: Reference dataset CANNOT be inclued in the query datasets")
+    }
+    else {
+      cat(
+        "Imputing given query datasets\n",
+        "Reference dataset:\n",
+        paste("  ", reference, "\n"),
+        "Query datasets:\n",
+        paste("  ", as.character(queries), "\n")
+        )
     }
-    else{
-      cat('Imputing given query datasets\n',
-          'Reference dataset:\n',
-          paste('  ', reference, '\n'),
-          'Query datasets:\n',
-          paste('  ', as.character(queries)))
     }
-  }
-  results = list()
   
-  reference_cells = rownames(object@scale.data[[reference]]) # cells by genes
+  reference_cells <- rownames(object@scale.data[[reference]]) # cells by genes
   for (query in queries) {
-    query_cells = rownames(object@scale.data[[query]])
-    #find nearest neighbors for query cell in normed ref datasets
-    nn.k = get.knnx(object@H.norm[reference_cells,],object@H.norm[query_cells,], k=knn_k, algorithm='CR')
-    imputed_vals = sapply(1:nrow(nn.k$nn.index),function(n){ # for each cell in the target dataset:
-      weights = nn.k$nn.dist[n,]
-      weights = as.matrix(exp(-weights)/sum(exp(-weights)))
-      imp = object@raw.data[[reference]][,nn.k$nn.index[n,]] # genes by cells, genes are from reference dataset
+    query_cells <- rownames(object@scale.data[[query]])
+    # find nearest neighbors for query cell in normed ref datasets
+    nn.k <- get.knnx(object@H.norm[reference_cells, ], object@H.norm[query_cells, ], k = knn_k, algorithm = "CR")
+    imputed_vals <- sapply(1:nrow(nn.k$nn.index), function(n) { # for each cell in the target dataset:
+      weights <- nn.k$nn.dist[n, ]
+      weights <- as.matrix(exp(-weights) / sum(exp(-weights)))
+      imp <- object@raw.data[[reference]][, nn.k$nn.index[n, ]] # genes by cells, genes are from reference dataset
       if (weight) {
-        imp = as.matrix(imp%*%weights) # (genes by k) multiply by the weight matrix (k by 1)
+        imp <- as.matrix(imp %*% weights) # (genes by k) multiply by the weight matrix (k by 1)
       }
       else {
-        imp = as.matrix(rowMeans(imp)) #simply count the rowmeans
+        imp <- as.matrix(rowMeans(imp)) # simply count the rowmeans
       }
       return(imp)
     })
-    colnames(imputed_vals) = query_cells
-    rownames(imputed_vals) = rownames(object@raw.data[[reference]])
-    object@raw.data[[query]] = imputed_vals
+    colnames(imputed_vals) <- query_cells
+    rownames(imputed_vals) <- rownames(object@raw.data[[reference]])
+    object@raw.data[[query]] <- imputed_vals
+  }
+  if (norm == TRUE) {
```

Comment:
Let's keep the convention of checking booleans with `if (bool)` (as you did for weight above). Same for the scale check below!

Reply:
Sorry for the late reply. I have fixed the documantation, warning and coding style issues. For the changes on structure, in this stage I am trying to avoid any changes which result in conflicts with former LIGER objects and functions, for example the incompatibility between `normalize`, `scaleNotCenter`,  `wilcoxon` functions and the imputed datasets. I will also fix the `if (bool)` problem soon.